independent-projects/tools/common/src/main/java/io/quarkus/devtools/project/extensions/ExtensionsManager.java

```diff
@@ -0,0 +1,71 @@
+package io.quarkus.devtools.project.extensions;
+
+import static io.quarkus.devtools.project.extensions.Extensions.equalsIgnoringVersions;
+
+import io.quarkus.dependencies.Extension;
+import io.quarkus.devtools.project.BuildTool;
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This interface defines a high level way of managing (read/write) extensions in any QuarkusProject
+ */
+public interface ExtensionsManager {
+
+    /**
+     * @return the {@link BuildTool} of this extension manager
+     */
+    BuildTool getBuildTool();
+
+    /**
+     * Read the build file(s) to get the list of extensions installed in this Quarkus project.
+     *
+     * @return The list of extensions installed in the project build file(s).
+     * @throws IOException if a problem occurs while reading the project build file(s)
+     */
+    List<Extension> read() throws IOException;
+
+    /**
+     * Read build file(s) to check if an extension is installed in this Quarkus project.
+     *
+     * @param e the extensions to check
+     * @return true if it's installed
+     * @throws IOException if a problem occurs while reading the project build file(s)
+     */
+    default boolean hasExtension(Extension e) throws IOException {
+        return read().stream().anyMatch(i -> equalsIgnoringVersions(i, e));
+    }
+
+    /**
+     * Check that the Quarkus Platform bom is defined in the project.
+     * The Quarkus Platform Bom role is to define version to use for Quarkus extensions.
+     *
+     * @return true if it's defined, false else
+     * @throws IOException if a problem occurs while reading the project build file(s)
+     */
+    boolean hasQuarkusPlatformBom() throws IOException;
+
+    /**
+     * This is going to add all the specified extensions to the project build file(s).
+     *
+     * Extensions which are already installed are ignored.
+     *
+     * @param extensions the list of extensions to add
+     * @return the number of added extensions (excluding already installed)
+     * @throws IOException if a problem occurs while reading/writing the project build file(s)
+     */
+    int add(List<Extension> extensions) throws IOException;
+
+    /**
+     * This is going to remove all the specified extensions from the project build file(s).
+     *
+     * This is ignoring the {@link Extension} version
+     *
+     * @param extensions the set of extensions to remove
+     * @return the number of removed extensions (excluding already not installed)
+     * @throws IOException if a problem occurs while reading/writing the project build file(s)
+     */
+    int remove(Set<Extension> extensions) throws IOException;
```

Comment:
Do they have to be `Extension` or the `AppArtifactKey` is enough? Also, perhaps it would be nice to use `Collection` instead

Reply:
Note that I renamed the different methods since your comment (install, getInstalled, uninstall..), I wanted to do it before to push, but I rushed a bit and forgot..

@gastaldi this is a good question, TBH I am not sure, what I think is important is to make sure that the input is well formatted, and not just some random string (that are resolved) like we get it the commands. At the end, for install, we need either the GA or the GAV (if the extension is not managed) and for uninstall just the GA, for `getInstalled` the GAV

Regarding `Set` or `Collection`, it's going to be converted to a Set anyway to remove duplicates..

Reply:
BTW, even if the extension is present in the platform, a user should still be able to override its version.

Reply:
@aloubyansky I am not 100% sure, but I think we said that since it's not a "recommended" action, we would advise the user to do it manually?

A possible middle group would be to when the specified version is different from the platform one, we add it, but show a warning?

Reply:
I don't think it costs us anything to respect the version choice.

Reply:
it is recommended to use a specific version when that is what is needed for the users app.
Thus if user are being explicit we should carry that information through the system to honor it.

Reply:
ok user version is now taken into account and I used `AppArtifactKey`